From 15a90fd9ac1396015340e599e26d7cd193898fb8 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Tue, 12 Aug 2014 14:26:13 +0800
Subject: [PATCH] configure.ac: add option --enable-thread-tls to manage thread
 ssl support
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The thread local storage caused arm-gcc broken while compiling                          │
syslog-ng with option '-g -O'.                                                          │
...                                                                                     │
dnscache.s: Assembler messages:                                                         │
dnscache.s:100: Error: invalid operands (.text and *UND* sections) for `-'              │
...                                                                                     │
                                                                                        │
Add option --enable-thread-tls to manage the including of thread
local storage, so we could explicitly disable it.

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>

change default to 'yes'
Upstream-Status: Submitted [https://github.com/syslog-ng/syslog-ng/pull/3649]

Signed-off-by: Yi Fan Yu <yifan.yu@windriver.com>
---
 configure.ac | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/configure.ac b/configure.ac
index 1d67e81..7aad75f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -210,6 +210,8 @@ AC_ARG_WITH(sanitizer,
               [  --with-sanitizer=[address/undefined/etc...]
                                          Enables compiler sanitizer supports (default: no)]
               ,,with_sanitizer="no")
+AC_ARG_ENABLE(thread-tls,
+              [  --enable-thread-tls        Enable Thread Local Storage support (default: yes)],,enable_thread_tls="yes")
 
 AC_ARG_ENABLE(dynamic-linking,
               [  --enable-dynamic-linking        Link everything dynamically.],,enable_dynamic_linking="auto")
@@ -628,12 +630,14 @@ dnl ***************************************************************************
 dnl Is the __thread keyword available?
 dnl ***************************************************************************
 
-AC_LINK_IFELSE([AC_LANG_PROGRAM(
-[[#include <pthread.h>
-__thread int a;
-]],
-[a=0;])],
-[ac_cv_have_tls=yes; AC_DEFINE_UNQUOTED(HAVE_THREAD_KEYWORD, 1, "Whether Thread Local Storage is supported by the system")])
+if test "x$enable_thread_tls" = "xyes"; then
+    AC_LINK_IFELSE([AC_LANG_PROGRAM(
+    [[#include <pthread.h>
+    __thread int a;
+    ]],
+    [a=0;])],
+    [ac_cv_have_tls=yes; AC_DEFINE_UNQUOTED(HAVE_THREAD_KEYWORD, 1, "Whether Thread Local Storage is supported by the system")])
+fi
 
 dnl ***************************************************************************
 dnl How to do static linking?
